src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -10061,6 +10061,10 @@ int RGWRados::check_bucket_shards(const RGWBucketInfo& bucket_info,
       return 0;
   }
 
+  if (bucket_info.is_indexless()) {
+    return 0;
+  }
+
   bool need_resharding = false;
   uint32_t num_source_shards = rgw::current_num_shards(bucket_info.layout);
```

Comment:
okay, but a suggestion for future-proofing:

`rgw::current_num_shards()` is the asserting version because the `layout` type is unchecked. instead of checking only for indexless here, can we reproduce the check that makes it safe to use the non-asserting overload? for example:

```c++
const auto& current_index = bucket_info.layout.current_index;
if (current_index.layout.type != BucketIndexType::Normal) {
  return 0;
}
uint32_t num_source_shards = rgw::num_shards(current_index.layout.normal);
```

that way we won't assert here on new index types